net/dns/recursive/recursive_test.go

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+package recursive
+
+import (
+	"context"
+	"math/rand"
+	"testing"
+	"time"
+
+	"tailscale.com/envknob"
+)
+
+func init() {
+	rand.Seed(time.Now().UnixMicro())
```

Comment:
We had a test recently that was dependent upon random seeding and it was flaking for some cases, with no control over it. I'm thinking about adding a package to make this easier, but in the meantime could you just avoid the seed here?

Reply:
Huh! Sure, happy to do that; this test won't fail no matter what the random seed is, since it's only used to shuffle lists, but happy to remove it since it's also not necessary.